homework_4/event_settler/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/jelenanemcic/code-cadets-2021/homework_4/event_settler/models"
+	"github.com/pkg/errors"
+	"io"
+	"log"
+	"math/rand"
+	"net/http"
+)
+
+func main() {
+	var settled []string
+
+	response, err := http.Get("http://127.0.0.1:8081/bets?status=active")
+	if err != nil {
+		log.Fatal(errors.WithMessage(err, "error in HTTP GET request"))
+	}
+
+	defer response.Body.Close()
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		log.Fatal(errors.WithMessage(err, "error reading body of HTTP response"))
+	}
+
+	var bets []models.BetReduced
+	err = json.Unmarshal(body, &bets)
+	if err != nil {
+		log.Fatal(errors.WithMessage(err, "error unmarshalling response body"))
+	}
+
+	for _, bet := range bets {
+		if !contains(settled, bet.SelectionId) {
+			var outcome string
+			if rand.Float64() > 0.5 {
+				outcome = "lost"
+			} else {
+				outcome = "won"
+			}
```

Comment:
This gets the job done but a cleaner and more maintainable solution would be to have an array of possible outcomes (can be hardcoded and would only have "lost" and "won" elements at the moment) and then selecting a random element from that array.